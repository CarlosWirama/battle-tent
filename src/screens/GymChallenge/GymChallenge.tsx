import React, { useContext, useEffect } from 'react';
import { Button, Centered } from '../../components/basics';
import { NavigationContext, ScreenState } from '../../navigation';
import { LayoutContainer } from './GymChallenge.styled';
import audio from '../../audio';
import {
  subscribeJoinedTheRoom,
  emitChallengeGymLeader,
} from '../../api';
import Navbar from '../../components/Navbar';
import Banner from '../../components/Banner';

const GymBgm = require('../../assets/audio/bgm/25_Pokemon_Gym.mp3');

export default function GymChallenge() {
  const navigate = useContext(NavigationContext);

  useEffect(() => {
    audio.playBgm(GymBgm)
  }, []);

  // subscriptions
  useEffect(function subscribe() {
    const sJoinedTheRoom = subscribeJoinedTheRoom(() => {
      navigate(ScreenState.Room);
    });
    return function unsubscribe() {
      sJoinedTheRoom.off();
    }
  }, [navigate]);

  return (
    <LayoutContainer>
      <header style={{ marginTop: 80 }}>
        <Navbar />
        <Centered>
          <Banner style={{ marginBottom: 20 }}>Gym Challenge</Banner>
        </Centered>
      </header>
      <main style={{ flex: 1 }}>
        <Button onClick={() => emitChallengeGymLeader(0)}>Pewter</Button>
        <Button onClick={() => emitChallengeGymLeader(1)}>Celadon</Button>
        <Button onClick={() => navigate(ScreenState.Lobby)}>Back</Button>
      </main>
    </LayoutContainer >
  );
}
