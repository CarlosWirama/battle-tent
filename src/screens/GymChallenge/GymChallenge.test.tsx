import React from 'react';
import { render } from '@testing-library/react';
import GymChallenge from './GymChallenge';

describe('GymChallenge screen', () => {
  it('should render', () => {
    const { asFragment } = render(
      <GymChallenge />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
