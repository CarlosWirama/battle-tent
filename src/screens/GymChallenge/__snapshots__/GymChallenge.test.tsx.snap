// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GymChallenge screen should render 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb sc-fzqMdD aTPwe"
  >
    <header
      style="margin-top: 80px;"
    >
      <nav
        class="sc-AxmLO kzsFQl"
      >
        <button
          aria-label="volume"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <img
              alt="music-volume"
              src="sound.png"
              width="28px"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </nav>
      <div
        class="sc-AxhCb kCMqZl"
      >
        <div
          class="sc-fzplWN ibMnsj"
          style="margin-bottom: 20px;"
        >
          Gym Challenge
        </div>
      </div>
    </header>
    <main
      style="flex: 1;"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Pewter
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cerulean
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Vermilion
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Celadon
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Fuchsia
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Saffron
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cinnabar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Viridian
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxirZ kOlOeA"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Back
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
